klippy/extras/tmc.py

```diff
@@ -230,7 +241,7 @@ def get_phase(self):
         return mscnt >> self.fields.get_field("MRES")
 
 # Helper to configure "stealthchop" mode
-def TMCStealthchopHelper(config, mcu_tmc, tmc_freq):
+def TMCStealthchopHelper(config, mcu_tmc, tmc_freq, tmc_type=2130):
```

Comment:
It's not necessary to pass the tmc_type to TMCStealthchopHelper as it already auto-detects the type.

Reply:
The current version does:
```python
    reg = fields.lookup_register("en_pwm_mode", None)
    if reg is not None:
        fields.set_field("en_pwm_mode", en_pwm_mode)
    else:
        # TMC2208 uses en_spreadCycle
        fields.set_field("en_spreadCycle", not en_pwm_mode)
```

my change was:
```python
    if tmc_type == 2130 or tmc_type == 5160:
        fields.set_field("en_pwm_mode", en_pwm_mode)
    elif tmc_type == 2208 or tmc_type == 2209:
        # TMC2208/2209 uses en_spreadCycle
        fields.set_field("en_spreadCycle", not en_pwm_mode)
```
is switched to tmc_type to have it consistent to the endstop method. I thought about the tmcs to come. We allways need to spreadcycle if threeshold is 0 in the future.

you want me to revert it?

Reply:
I think it would be preferably if both TMCStealthchopHelper and TMCEndstopHelper did auto-detection, and neither was passed a tmc_type.

Reply:
changed to if reg solution